analyzers/src/SonarAnalyzer.VisualBasic/Rules/ClassShouldNotBeEmpty.cs

```diff
@@ -0,0 +1,35 @@
+﻿/*
+ * SonarAnalyzer for .NET
+ * Copyright (C) 2015-2023 SonarSource SA
+ * mailto: contact AT sonarsource DOT com
+ *
+ * This program is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU Lesser General Public
+ * License as published by the Free Software Foundation; either
+ * version 3 of the License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with this program; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+ */
+
+namespace SonarAnalyzer.Rules.VisualBasic;
+
+[DiagnosticAnalyzer(LanguageNames.VisualBasic)]
+public sealed class ClassShouldNotBeEmpty : ClassShouldNotBeEmptyBase<SyntaxKind>
+{
+    protected override ILanguageFacade<SyntaxKind> Language => VisualBasicFacade.Instance;
+
+    protected override bool IsEmptyAndNotPartial(SyntaxNode node) =>
+        node is ClassBlockSyntax { Members.Count: 0 } classSyntax
+        && !classSyntax.ClassStatement.Modifiers.Any(x => x.IsKind(SyntaxKind.PartialKeyword));
+
+    protected override bool IsClassWithDeclaredBaseClass(SyntaxNode node) => node is ClassBlockSyntax { Inherits.Count: > 0 };
+
+    protected override string DeclarationTypeKeyword(SyntaxNode node) => ((TypeBlockSyntax)node).BlockStatement.DeclarationKeyword.ValueText.ToLower();
```

Comment:
[safety check] is this cast safe?

Reply:
Changed it by adding type check and a default value for both C# and VB implementations.

Reply:
can you please add a message in the UT for this case? (it seems it's covered by tests already)

Reply:
It wasn't up to date. When the type check is added, the false branch never gets hit (even if the declared type is missing / compiler error). So it's only a fallback for future development.
I fixed the UT.

Reply:
Ok, I checked myself and the answer to my initial question is: yes, the cast is safe because the rules only register for type declaration syntaxes because:
- all C# type declarations (ClassDeclarationSyntax, InterfaceDeclarationSyntax, StructDeclarationSyntax, RecordDeclarationSyntaxWrapper) inherit from TypeDeclarationSyntax
- all VB .NET type declarations `ClassBlockSyntax` inherits from TypeBlockSyntax

It does not make sense to be defensive when the type system is clear.

And, if in doubt, the pattern in the codebase is to trust the type system:
- GenericInheritanceShouldNotBeRecursive - see SyntaxKinds and the inheritance `GenericInheritanceShouldNotBeRecursiveBase<SyntaxKind, TypeDeclarationSyntax>` 
- and inside GenericInheritanceShouldNotBeRecursiveBase we do the cast `var declaration = (TDeclaration)c.Node;`

Reply:
Removed the type checks.